src/index.jsx

```diff
@@ -24,13 +25,17 @@ import * as viewutils from './utils/view'
 
 let dataPackage
 
-dputils.fetchDataPackageOnly(DATA_PACKAGE_URL).then(async dpObj => {
+let divElements = document.querySelectorAll('.react-me')
+
+fetchDataPackageAndDataIncrementally(DATA_PACKAGE_URL, divElements)
+
+async function fetchDataPackageAndDataIncrementally(dataPackageIdentifier, divElements) {
```

Comment:
I'm still not sure we've got the code architecture quite right here.

For now we can leave.

However, my sense is we will want to refactor to have a cleaner separation of:

- Data loading with callback on each new piece of data
- Event handling of that event with relevant dispatch to dependent objects.

This will make refactoring to redux later super easy and current code comprehensible and easy to maintain.